src/sourmash/sourmash_args.py

```diff
@@ -392,15 +392,22 @@ def load_pathlist_from_file(filename):
     "Load a list-of-files text file."
     try:
         with open(filename, 'rt') as fp:
-            file_list = [ x.rstrip('\r\n') for x in fp ]
-
-        if not os.path.exists(file_list[0]):
-            raise ValueError("first element of list-of-files does not exist")
+            temp_file_list = [ x.rstrip('\r\n') for x in fp ]
+        file_list = set(temp_file_list)
+        if not file_list:
+            raise ValueError("pathlist is empty")
+        badFormat = False
+        for checkfile in file_list:
+            if not os.path.exists(checkfile):
+                badFormat = True
+        if badFormat:
+            raise ValueError("pathlist contains a badly formatted file")
+    except IOError:
+        raise ValueError(f"file '{filename}' does not exist")    
```

Comment:
hmm, I'm confused by the error messages you chose here -

1. if the file *inside* the pathlist doesn't exist, the error message is `pathlist contains a badly formatted file`
2. if the pathlist file itself doesn't exist, the error message is "file does not exist".

A suggestion --

* if the file inside the pathlist file doesn't exist, the error message could be "file X listed inside pathlist Y does not exist"
* if the pathlist file itself doesn't exist (the IOError), the error message could be "pathlist file 'filename' doesn't exist"

Reply:
In this case, should I remove `badformat` altogether? Should I just raise the error inside the for loop? I could write something like this:
`for checkfile in file_list:`
`if not os.path.exists(checkfile):`
`raise ValueError("pathlist contains a badly formatted file")`